src/ir/auth_logic/ast.h

```diff
@@ -295,31 +347,34 @@ class SaysAssertion {
     return visitor.Visit(*this);
   }
 
-  // A potentially ugly print of the state in this class
-  // for debugging/testing only
-  std::string DebugPrint() const {
-    std::vector<std::string> assertion_strings;
-    assertion_strings.reserve(assertions_.size());
-    for (const Assertion& assertion : assertions_) {
-      assertion_strings.push_back(assertion.DebugPrint());
+  bool operator==(const SaysAssertion& rhs) const {
+    return principal_ == rhs.principal() && assertions_ == rhs.assertions();
+  }
+  virtual bool operator==(const ASTNode& rhs) const override {
+    try {
+      const SaysAssertion & rhsSays = dynamic_cast<const SaysAssertion&>(rhs);
+      return *this == rhsSays;
+    } catch(const std::bad_cast& e) {
+      return false;
```

Comment:
Don't use a try catch for  this.

Reply:
@markww Just to save yourself some typing in the future, you could try writing comments to express a general pattern in one place and requesting me to do this "throughout the code". Of course you can continue to keep doing it this way if you prefer.